### DISCUSSION TITLE
CNC Board with PWM laser will not work.

### DISCUSSION THREAD

**Discussion by timtiernan** [arkypita/LaserGRBL#1069] [Q&A]
Hi, I've tried everything I can think of but I'm really stuck! You're my last hope.

I have recently purchased a new CNC board (https://www.amazon.co.uk/dp/B07D211NKD/ref=cm_sw_r_cp_apa_i_FC5aFbWY4T8NA) and laser (https://www.amazon.co.uk/dp/B07VD47BMX/ref=cm_sw_r_cp_apa_i_cD5aFbKE41RB9) but they just will not talk to eachother.

I've got it wired as shown on the page for the board but for some reason when I send the following commands the laser doesn't switch on (PWM is enabled in settings).

![15940706572334569275048136304277](https://user-images.githubusercontent.com/67929245/86646796-729d3280-bfd7-11ea-829c-43abe394571e.jpg)

I've tried it with and without the 12v external power to the driver board (adapter purchased separately (https://www.amazon.co.uk/dp/B017OMEW4Q/ref=cm_sw_r_cp_apa_i_RD5aFbARDEXH6)) but I'm not getting anything except the fans on the board and laser running when switched on (so I know it's getting power). I've also tried it with the jumper set to both laser and spindle with no effect. 

I don't believe that the board is faulty as it has s responding to LaserGRBL, just not doing anything on the other end.

I have had CNC boards and lasers work in the past but for some reason this one won't play ball.

Thanks in advance for any help offered.

![15940706065485694979831415665301](https://user-images.githubusercontent.com/67929245/86646649-55686400-bfd7-11ea-9501-11308eb28b61.jpg)

![15940707354184129086456267654002](https://user-images.githubusercontent.com/67929245/86647032-a37d6780-bfd7-11ea-8eed-ef49a1e912ff.jpg)

![15940707602963499351941549077198](https://user-images.githubusercontent.com/67929245/86647096-aed09300-bfd7-11ea-9b62-e0be311c2c4d.jpg)

**Comment by Maxoch** [comment]
Попробуйте подключить сначала только +12-red, GND - black, +5 - yellow  нажмите на кнопку на драйвере. Если лазер запустится, значит он в порядке. тогда подключайте лазер как показано на картинки и быдет вам счастье))
![86646649-55686400-bfd7-11ea-9501-11308eb28b61](https://user-images.githubusercontent.com/66250062/86723004-8dec5a00-c038-11ea-9451-9f05b255d32f.jpg)

**Comment by gmmanonymus111** [comment]
You seem to send the power on command after enabling laser mode ($31). You need to be "moving" in laser mode to be able to turn the laser on. Use `G1F1M3S5`, or turn off laser mode (`$31=0`) and use `M3S5`.
EDIT:
Maybe you need to replace the jumper from 'spn' to 'lsr' next to the middle driver. Also, are you sure your wiring is correct? The board you linked misses the 3-pin port where your laser is plugged.

**Comment by timtiernan** [comment]
> Попробуйте подключить сначала только +12-red, GND - black, +5 - yellow нажмите на кнопку на драйвере. Если лазер запустится, значит он в порядке. тогда подключайте лазер как показано на картинки и быдет вам счастье))
> ![86646649-55686400-bfd7-11ea-9501-11308eb28b61](https://user-images.githubusercontent.com/66250062/86723004-8dec5a00-c038-11ea-9451-9f05b255d32f.jpg)

Hi, I only have one GND out on the driver board.

![IMG20200708211005](https://user-images.githubusercontent.com/67929245/86965433-752a9400-c15f-11ea-99a3-33b10f02b0a3.jpg)

**Comment by timtiernan** [comment]
> You seem to send the power on command after enabling laser mode ($31). You need to be "moving" in laser mode to be able to turn the laser on. Use `G1F1M3S5`, or turn off laser mode (`$31=0`) and use `M3S5`.
> EDIT:
> Maybe you need to replace the jumper from 'spn' to 'lsr' next to the middle driver. Also, are you sure your wiring is correct? The board you linked misses the 3-pin port where your laser is plugged.

Hi,

I've tried enumerable different wiring options to no avail. It does seem to be right, but for some reason it's just not picking up the PWM message.

**Comment by Maxoch** [comment]
> Привет, у меня только один GND на плате водителя.

Вам нужно переделать кабель. Я на своей машине сделал именно так как на рисунке (1cable)
![Безымянный](https://user-images.githubusercontent.com/66250062/87012710-a85e3900-c1da-11ea-84a0-b1aca56426c5.png)

**Comment by StuartB4** [comment]
Like this. 

![107539837_10158712093203944_4532997456347331997_o](https://user-images.githubusercontent.com/16980026/87044743-e5342b00-c1ee-11ea-9a6f-6424933c48d3.jpg)

![Untitioioiled](https://user-images.githubusercontent.com/16980026/87044766-eb2a0c00-c1ee-11ea-80f0-9db95b8a618a.jpg)
